DataFormats/L1TrackTrigger/interface/TTTrack.h

```diff
@@ -205,137 +279,130 @@ void TTTrack<T>::setFitParNo(unsigned int nPar) {
 
 template <typename T>
 GlobalVector TTTrack<T>::momentum() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theMomentum;
-  } else
-    return GlobalVector(0.0, 0.0, 0.0);
-}
-
-template <typename T>
-GlobalVector TTTrack<T>::getMomentum(unsigned int npar) const {
-  return momentum();
+  return theMomentum_;
 }
 
 template <typename T>
 double TTTrack<T>::rInv() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theRInv;
-  } else
-    return 0.0;
-}
-
-template <typename T>
-double TTTrack<T>::getRInv(unsigned int npar) const {  //backwards compatibility
-
-  return rInv();
+  return theRInv_;
 }
 
 template <typename T>
 double TTTrack<T>::tanL() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theTanL;
-  } else
-    return 0.0;
+  return theTanL_;
 }
 
 template <typename T>
 double TTTrack<T>::eta() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theEta;
-  } else
-    return 0.0;
+  return theMomentum_.eta();
 }
 
 template <typename T>
 double TTTrack<T>::phi() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return thePhi;
-  } else
-    return 0.0;
+  return thePhi_;
 }
 
 template <typename T>
 double TTTrack<T>::d0() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theD0;
-  } else
-    return 0.0;
+  return theD0_;
 }
 
 template <typename T>
 double TTTrack<T>::z0() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theZ0;
-  } else
-    return 0.0;
+  return theZ0_;
 }
 
 template <typename T>
 GlobalPoint TTTrack<T>::POCA() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return thePOCA;
-  } else
-    return GlobalPoint(0.0, 0.0, 0.0);
+  return thePOCA_;
 }
 
+/// Chi2
 template <typename T>
-GlobalPoint TTTrack<T>::getPOCA(unsigned int npar) const  //backwards compatibility
-{
-  return POCA();
+double TTTrack<T>::chi2() const {
+  return theChi2_;
 }
 
-/// Chi2
+/// Chi2Z
 template <typename T>
-double TTTrack<T>::chi2() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theChi2;
-  } else
-    return 0.0;
+double TTTrack<T>::chi2Z() const {
+  return theChi2_Z_;
 }
 
+/// Chi2XY
 template <typename T>
-double TTTrack<T>::getChi2(unsigned int npar) const  //backwards compatibility
-{
-  return chi2();
+double TTTrack<T>::chi2XY() const {
+  return theChi2_XY_;
 }
 
 /// Chi2 reduced
 template <typename T>
 double TTTrack<T>::chi2Red() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theChi2 / (2 * theStubRefs.size() - NumFitPars);
-  } else
-    return 0.0;
+  return theChi2_ / (2 * theStubRefs.size() - theNumFitPars_);
+}
+
+/// Chi2XY reduced
+template <typename T>
+double TTTrack<T>::chi2XYRed() const {
+  return theChi2_XY_ / (theStubRefs.size() - theNumFitPars_ - 2);
```

Comment:
```suggestion
  return theChi2_XY_ / (theStubRefs.size() - (theNumFitPars_ - 2));
```
@mdhildreth, @tomalin suggest to make this fix